File: templates/mobile/android/app/src/main/java/glyphspace/host/GlyphspaceRuntimeBridge.kt
package glyphspace.host

class GlyphspaceRuntimeBridge {
    var acceptedPatchCount: Int = 0
        private set

    fun loadWorld(bytes: ByteArray): Boolean = bytes.isNotEmpty()

    fun applyPatch(bytes: ByteArray): Boolean {
        if (bytes.isEmpty()) return false
        acceptedPatchCount += 1
        return true
    }

    fun accessibilityLabel(glyphId: String, fallback: String): String =
        "$fallback, Glyphspace glyph $glyphId"
}

File: templates/mobile/ios/Sources/GlyphspaceMobile/GlyphspaceRuntimeBridge.swift
import Foundation
import UIKit

public final class GlyphspaceRuntimeBridge {
    public private(set) var acceptedPatchCount: Int = 0

    public init() {}

    public func loadWorld(_ bytes: Data) -> Bool {
        !bytes.isEmpty
    }

    public func applyPatch(_ bytes: Data) -> Bool {
        guard !bytes.isEmpty else { return false }
        acceptedPatchCount += 1
        return true
    }

    public func accessibilityLabel(for glyphId: String, fallback: String) -> String {
        "\(fallback), Glyphspace glyph \(glyphId)"
    }
}
